FileSystemSimulator/FileSystem/FileSystem.cpp

```diff
@@ -48,6 +48,25 @@ bool FileSystem::create(const std::string & i_file_name)
   return true;
 }
 
+bool FileSystem::destroy(const std::string & i_file_name)
+{
+	if (i_file_name == "")
+		return false;
+	
+	DirEntry* entry = findDirEntry(i_file_name);
+	if (!entry) 
+		return false;
+	string destroyed_file_name = "";
+	std::memcpy(entry->file_name, destroyed_file_name.c_str(), DirEntry::MAX_FILE_NAME_LENGTH);
+	
+	FileDescriptor fd = getFileDescriptor(entry->file_descr_index);
+	for (int i = 0; i <= fd.getLastBlockIndex; i++) {
+		setBit(fd.data_blocks[i], true);
+	}
+	fd.is_free = true;
```

Comment:
```suggestion
	fd.is_free = true;
	fd.clearDataBlocks();
	writeFileDescriptorToIO(fd, entry->file_descr_index);
```
need to write modified file descriptor on disk, fd - just local variable
clearDataBlocks() set -1 to data_blocks (means they are not allocated)